src/main/java/de/hysky/skyblocker/skyblock/waypoint/WaypointsListWidget.java

```diff
@@ -0,0 +1,291 @@
+            return children;
+        }
+
+        @Override
+        public List<? extends Element> children() {
+            return children;
+        }
+
+        private void updateName(String name) {
+            int index = waypoints.indexOf(category);
+            category = category.withName(name);
+            if (index >= 0) {
+                waypoints.set(index, category);
+            }
+        }
+
+        @Override
+        public void render(DrawContext context, int index, int y, int x, int entryWidth, int entryHeight, int mouseX, int mouseY, boolean hovered, float tickDelta) {
+            enabled.setPosition(x, y + 1);
+            nameField.setPosition(x + 22, y);
+            buttonNewWaypoint.setPosition(x + entryWidth - 115, y);
+            buttonDelete.setPosition(x + entryWidth - 38, y);
+            for (ClickableWidget child : children) {
+                child.render(context, mouseX, mouseY, tickDelta);
+            }
+        }
+    }
+
+    protected class WaypointEntry extends AbstractWaypointEntry {
+        private final WaypointCategoryEntry category;
+        private NamedWaypoint waypoint;
+        private final List<ClickableWidget> children;
+        private final CheckboxWidget enabled;
+        private final TextFieldWidget nameField;
+        private final TextFieldWidget xField;
+        private final TextFieldWidget yField;
+        private final TextFieldWidget zField;
+        private final TextFieldWidget colorField;
+        private final ButtonWidget buttonDelete;
+
+        public WaypointEntry(WaypointCategoryEntry category) {
+            this(category, new NamedWaypoint(BlockPos.ORIGIN, "New Waypoint", new float[]{0f, 1f, 0f}));
+        }
+
+        public WaypointEntry(WaypointCategoryEntry category, NamedWaypoint waypoint) {
+            this.category = category;
+            this.waypoint = waypoint;
+            enabled = CheckboxWidget.builder(Text.literal(""), client.textRenderer).checked(screen.isEnabled(waypoint)).callback((checkbox, checked) -> screen.enabledChanged(waypoint, checked)).build();
+            nameField = new TextFieldWidget(client.textRenderer, 65, 20, Text.literal("Name"));
+            nameField.setText(waypoint.getName().getString());
+            nameField.setChangedListener(this::updateName);
+            xField = new TextFieldWidget(client.textRenderer, 26, 20, Text.literal("X"));
+            xField.setText(Integer.toString(waypoint.pos.getX()));
+            xField.setChangedListener(this::updateX);
+            yField = new TextFieldWidget(client.textRenderer, 26, 20, Text.literal("Y"));
+            yField.setText(Integer.toString(waypoint.pos.getY()));
+            yField.setChangedListener(this::updateY);
+            zField = new TextFieldWidget(client.textRenderer, 26, 20, Text.literal("Z"));
+            zField.setText(Integer.toString(waypoint.pos.getZ()));
+            zField.setChangedListener(this::updateZ);
```

Comment:
Using #setTextPredicate to prevent from typing invalid numbers could be neat

Reply:
Done in ee6664d.